modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/wal/filehandle/FileWriteHandleImpl.java

```diff
@@ -611,4 +586,137 @@ public String safePosition() {
     @Override public int getSwitchSegmentRecordOffset() {
         return switchSegmentRecordOffset;
     }
+
+    /**
+     * Interface for performing {@link MappedByteBuffer} fsync.
+     */
+    private interface MMapFSyncer {
+        /** {@link MappedByteBuffer#fd} */
+        static final Field fd = findField(MappedByteBuffer.class, "fd");
+
+        /**
+         * Performs fsync.
+         *
+         * @param buf Mmapped byte buffer.
+         * @param index Index of the syncing segment in the buffer.
+         * @param len Length of the syncing segment part.
+         * @throws IgniteCheckedException If failed.
+         */
+        void fsync(MappedByteBuffer buf, int index, int len) throws IgniteCheckedException;
+    }
+
+    /**
+     * @return FSyncer suitable for the current JRE.
+     */
+    private static MMapFSyncer pickFsyncer() {
+        int javaVersion = majorJavaVersion(jdkVersion());
+
+        if (javaVersion >= 15)
+            return new JDK15FSyncer();
+
+        return new LegacyFSyncer();
+    }
+
+    /**
+     * Runs fsync operation on JRE15 and higher using MappedMemoryUtils which provides aligned fsync.
+     */
+    private static class JDK15FSyncer implements MMapFSyncer {
```

Comment:
Does this syncer perform better than the legacy one? Its code does not seem much simpler, but now we have 2 syncers to support.

Reply:
The old one doesn't work on JDK15 and newer

Reply:
Oh, the most obvious one :)